0.preprocess-sites/8.visualize-plate-layout.py

```diff
@@ -0,0 +1,392 @@
+import os
+import sys
+import pathlib
+import argparse
+import pandas as pd
+import matplotlib.pyplot as plt
+import seaborn as sns
+
+# import numpy as np
+
+sys.path.append(os.path.join("..", "scripts"))
+from config_utils import process_config_file
+from cell_quality_utils import CellQuality
+
+parser = argparse.ArgumentParser()
+parser.add_argument(
+    "--config_file",
+    help="configuration yaml file for preprocessing pipeline",
+    default="site_processing_config.yaml",
+)
+args = parser.parse_args()
+config_file = args.config_file
+
+config = process_config_file(config_file)
+
+# Defines the sections of the config file
+core_args = config["core"]
+spots_args = config["process-spots"]
+summ_cell_args = config["summarize-cells"]
+summ_plate_args = config["summarize-plate"]
+
+# Defines the variables set in the config file
+batch = core_args["batch"]
+quality_func = core_args["categorize_cell_quality"]
+ignore_files = core_args["ignore_files"]
+acquisition_side = core_args["acquisition_side"]
+batch_dir = core_args["batch_dir"]
+
+cell_filter = spots_args["cell_filter"]
+
+summ_cells_results_basedir = summ_cell_args["output_resultsdir"]
+summ_cells_figures_basedir = summ_cell_args["output_figuresdir"]
+output_figuresdir = pathlib.Path(summ_cells_figures_basedir, batch)
+cell_category_order = summ_cell_args["cell_category_order"]
+
+correlation_threshold = summ_plate_args["correlation_threshold"]
+painting_image_names = summ_plate_args["painting_image_names"]
+barcoding_cycles = summ_plate_args["barcoding_cycles"]
+barcoding_prefix = summ_plate_args["barcoding_prefix"]
+
+cell_count_file = pathlib.Path(
+    summ_cells_results_basedir, batch, "cells", "cell_count.tsv"
+)
+cell_count_df = pd.read_csv(cell_count_file, sep="\t")
+
+figures_output = pathlib.Path(summ_cells_figures_basedir, batch)
+os.makedirs(figures_output, exist_ok=True)
+results_output = pathlib.Path(summ_cells_results_basedir, batch)
+os.makedirs(results_output, exist_ok=True)
+
+# creates x, y coordinates for plotting per-plate views.
+# assumes image numbering starts in upper left corner and proceeds down
+x_seq = 1
+y_seq = 1
+build_seq = []
+final_order = []
+for i in range(1, ((acquisition_side * acquisition_side) + 1)):
+    if len(final_order) == (acquisition_side * acquisition_side):
+        break
+    else:
+        build_seq.append([x_seq, y_seq])
+        if len(build_seq) == acquisition_side:
+            x_seq = x_seq + 1
+        if y_seq == acquisition_side:
+            y_seq = 0
+            build_seq = build_seq[::-1]
+            final_order.extend(build_seq)
+            build_seq = []
+        y_seq = y_seq + 1
+sites_list = [*range(1, (acquisition_side * acquisition_side) + 1)]
+
+# uses sites_list in case there are fewer analyzed sites than acquired sites
+loc_df = pd.DataFrame(final_order).rename(columns={0: "x_loc", 1: "y_loc"})
+loc_df["Site"] = sites_list
```

Comment:
```suggestion
loc_df = pd.DataFrame(final_order).rename(columns={0: "x_loc", 1: "y_loc"}).assign(site=sites_list)
```

Also note the lowercase suggestion (see comment below)